apps/system/js/base_module.js

```diff
@@ -110,7 +102,83 @@
    */
   BaseModule.STATES = [];
 
+  BaseModule.SIDE_MODULES = [];
+
+  var SidemoduleMixin = {
+    _startSideModules: function() {
+      if (!this.constructor.SIDE_MODULES ||
+          this.constructor.SIDE_MODULES.length === 0) {
+        this.debug('no side modules. back.');
+        return;
+      }
+
+      var submodules = this.constructor.SIDE_MODULES.slice();
+      var unloaded = [];
+      submodules.forEach(function(submodule) {
+        if (BaseModule.defined(submodule) || window[submodule]) {
+          var name = BaseModule.lowerCapital(submodule);
+          if (!this[name]) {
+            this._initialSideModule(name, submodule);
+          }
+        } else {
+          unloaded.push(submodule);
+        }
+      }, this);
+
+      if (unloaded.length === 0) {
+        this.__side_module_loaded && this.__side_module_loaded();
+        return;
+      }
+
+      this.debug('lazy loading side modules: ' +
+        unloaded.concat());
+      BaseModule.lazyLoad(unloaded).then(function() {
+        this.debug('lazy loaded side modules: ' +
+          unloaded.concat());
+        unloaded.forEach(function(module) {
+          var moduleName = BaseModule.lowerCapital(module);
+          if (!this[moduleName]) {
+            this._initialSideModule(moduleName, module);
+          }
+        }, this);
+        this.__side_module_loaded && this.__side_module_loaded();
+      }.bind(this)).catch(function(err) {
+        console.error(err);
+      });
```

Comment:
this code is pretty hard to read and _very_ similar to the sub module code, we should factor it :)